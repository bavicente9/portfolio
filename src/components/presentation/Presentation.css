@import url('https://fonts.googleapis.com/css2?family=Indie+Flower&display=swap');

.presentation {
    background: var(--GeneralBackGround);
}

.presentation__container {
    padding: 50px var(--GeneralMarginWidth) ;
    display: grid;
    grid-template-areas:
        'title avatar'
        'description avatar'
        'buttons avatar'
    ;
    grid-template-columns: 50% 50%;
    grid-template-rows: min-content min-content min-content;

    place-items: center;
    color: #fff;
}

/*Info*/
.presentation h1 {
    grid-area: title;
    width: 100%;
    max-width: 650px;

    margin-bottom: 0;

    font-family: 'Indie Flower', cursive, sans-serif, cursive;
    font-size: 2.5rem;

}

.presentation p {
    grid-area: description;
    width: 100%;
    max-width: 650px;
    font-size: 1.5rem;
}

.presentation__buttons {
    grid-area: buttons;
    display: flex;
    justify-content: left;
    align-items: center;
    gap: 25px;
}

.button-presentation {
    display: grid;
    place-items: center;
    text-decoration: none;
}

/*Image*/

.presentation__avatar {
    grid-area: avatar;
    height: 25vw;
    max-height: 300px;
    width: 20vw;
    max-width: 300px;
}

.presentation__avatar img {
    width: 100%;
    height: 100%;
    object-fit: contain;
}


@media (max-width:930px) {

    .presentation__container {
        display: grid;
        grid-template-areas:
            'title avatar'
            'description description'
            'buttons buttons'
        ;
        grid-template-columns: 50% 50%;
        grid-template-rows: min-content min-content min-content;

        justify-items:left;

        padding: 50px var(--GeneralRMarginWidth);
    }

    /*Info*/
    .presentation h1 {
        font-size: 2rem;
    }

    .presentation p {
        font-size: 1.2rem;
    }
    
   

    /*Image*/

    .presentation__avatar {
        min-height: 150px;
        min-width: 120px;
        justify-self: center;
    }

}