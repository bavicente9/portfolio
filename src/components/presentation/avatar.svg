<svg width="191.951" height="229.875" viewBox="0 0 50.787 60.821" xmlns="http://www.w3.org/2000/svg"><path d="m87.219 67.436-9.94 5.262c-.778.512-1.29 1.292-1.68 2.192l-5.92 15.42c-.034 3.858 1.794 5.684 3.874 7.235l42.68.074c3.027-1.403 4.354-4.128 3.873-8.259l-5.043-13.812c-.576-1.023-1.494-2.046-2.85-3.07l-9.5-5.188c-5.064 5.576-10.234 5.287-15.493.146z" fill="#3a89c5" stroke="#020000" stroke-width=".265" style="fill:#ce8600;fill-opacity:1" transform="translate(-69.547 -36.93)"/><rect x="78.23" y="76.278" width="33.106" height="21.194" ry="1.389" fill="#b6b6b6" stroke="#020000" stroke-width=".265" transform="translate(-69.547 -36.93)"/><g fill="#ffd1b7" stroke="#000"><path d="M15.767 24.507c-.194-6.055.03-.001-.194-6.055-2.207-.29-2.901.917-2.76 2.473.15 1.66 1.429 3.55 2.954 3.582z" stroke-width=".258"/><path d="M34.663 24.491c.22-6.606-.035-.001.22-6.606 2.518-.316 3.31 1.001 3.149 2.698-.172 1.81-1.63 3.872-3.369 3.908z" stroke-opacity=".944" stroke-width=".288"/></g><path d="M15.822 24.462c.89 9.273 7.402 10.665 9.509 10.705 4.152.078 9.865-4.564 9.611-10.705l-.103-9.198c-2.055-8.12-6.005-8.858-9.707-8.692-3.11.14-7.167-.253-9.413 8.485z" fill="#ffd1b7" stroke="#000" stroke-opacity=".924" stroke-width=".265"/><g transform="translate(-69.547 -36.93)"><rect x="86.576" y="55.328" width="6.845" height="5.691" ry="2.039" fill="#f9f9f9" stroke="#000" stroke-width=".318"/><rect x="96.258" y="55.429" width="6.845" height="5.691" ry="2.039" fill="#f9f9f9" stroke="#000" stroke-width=".318"/><rect x="93.243" y="56.982" width="3.152" height=".776" rx="0" ry=".388" fill-rule="evenodd"/></g><path d="M15.574 18.451c.053.098.068.144.054.148-.087.025-1.307-1.583-1.318-1.861-.352-2.1-.48-4.28-.107-6.718a9.61 9.61 0 0 1 .732-1.376 5.23 5.23 0 0 1 .883-1.036c-.453-2.203 2.244-4.055 4.01-5.26 1.07-.51 2.32-.38 3.633-.019l3.379 1.535c1.192.452 2.59.453 3.903-.031.669-.464.903-.991 1.238-1.594a7.179 7.179 0 0 0 .735-1.98c2.331 2.242 2.5 5.105 2.579 8.068 1.599 3.523 1.1 6.05.583 8.718-.288.561-.436 1.387-.923 1.315-.85-1.803-1.292-5.167-3.963-3.783-3.015.583-5.902 1.624-9.896.053-1.339-.51-2.719-2.292-4.186-.099-.3.683-.193 1.326-.4 1.91-.253.713-.54 1.51-.882 2.158z" stroke="#000" stroke-width=".265"/></svg>