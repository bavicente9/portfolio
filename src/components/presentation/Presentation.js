import './Presentation.css'
import img_avatar from './avatar.svg'

export const Presentation = () => {
    return (
        <section aria-label='Presentation' id='presentation' className='presentation'>
            <div className='presentation__container'>
                <h1>Hello! i'm Bryan Vicente, web developer.</h1>
                <p>I am passionate about technology,improve my skills , get new knowledge continuously, and build  the bests websites with good practics and a great accessibility.</p>
                <div className='presentation__buttons'>
                    <a rel='external' className='button button-presentation' href='https://github.com/bavicente9' >Resume</a>
                    <a rel='external' className='button button-presentation' href='https://github.com/bavicente9' >GitHub</a>
                </div>
                <div className='presentation__avatar'>
                    <img alt='' src={img_avatar} />
                </div>
            </div>
        </section>
    )
}