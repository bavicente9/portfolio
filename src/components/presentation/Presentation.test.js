import { render, screen } from '@testing-library/react'
import { Presentation } from "./presentation"


describe('Section: Presentation', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<Presentation />)
        expect(asFragment()).toMatchSnapshot()
    })
})