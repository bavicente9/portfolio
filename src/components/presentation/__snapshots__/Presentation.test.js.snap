// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section: Presentation renders correctly 1`] = `
<DocumentFragment>
  <section
    aria-label="Presentation"
    class="presentation"
    id="presentation"
  >
    <div
      class="presentation__container"
    >
      <h1>
        Hello! i'm Bryan Vicente, web developer.
      </h1>
      <p>
        I am passionate about technology,improve my skills , get new knowledge continuously, and build  the bests websites with good practics and a great accessibility.
      </p>
      <div
        class="presentation__buttons"
      >
        <a
          class="button button-presentation"
          href=""
          rel="external"
        >
          Resume
        </a>
        <a
          class="button button-presentation"
          href="https://github.com/bavicente9"
          rel="external"
        >
          GitHub
        </a>
      </div>
      <div
        class="presentation__avatar"
      >
        <img
          alt=""
          src="avatar.svg"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
